ext/standard/head.c

```diff
@@ -148,7 +148,12 @@ PHPAPI int php_setcookie(zend_string *name, zend_string *value, time_t expires,
 			strlcat(cookie, ZSTR_VAL(dt), len + 100);
 			zend_string_free(dt);
 
-			snprintf(tsdelta, sizeof(tsdelta), ZEND_LONG_FMT, (zend_long) difftime(expires, time(NULL)));
+			int diff;
+			diff = difftime(expires, time(NULL));
+			if (diff < 0) {
+				diff = 0;
```

Comment:
It's not a warning type event though is it? An expires at date in the past is valid, it's just that "max-age" shouldn't be negative